src/components/BookList.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import AddBook from './AddBook';
+
+const BookList = () => {
+  const [books, setBooks] = useState([]);
+  const delt = (id) => {
+    setBooks([...books.filter((book) => book.id !== id)]);
+  };
+  return (
+    <div>
+      <h2> LIST OF BOOKS </h2>
+      {' '}
+      {books.map((book) => (
+        <ul key={book.id}>
+          <li>
+            {' '}
+            Book Title:
+            {' '}
+            {book.title}
+            {' '}
+            | by:
+            {' '}
+            {book.author}
+            {' '}
+            <button type="submit" onClick={() => delt(book.id)}>
+              {' '}
```

Comment:
- [ ] Please as requested in the requirements there should be a component that  Contains individual book states, as well as a button to delete the book, as I can see you just put all the books logic in one component, kindly wrap the book in a component then render it in the map function as a list item.

Reply:
Hello  @MeqdamAlqudah , thank you for the feedback, 
please can you help me to understand this better, I have already spent a lot of hours on this particular section.